metrics/web/lib/base/presentation/widgets/base_popup.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:flutter/material.dart';
+
+/// Signature for the function that builds an offset using the [childSize].
+typedef OffsetBuilder = Offset Function(Size childSize);
+
+/// A widget that displays the given [child] that opens the given [popup]
+/// when is triggered.
+class BasePopup extends StatelessWidget {
+  /// A widget to display as a [popup] trigger.
+  final Widget child;
+
+  /// A widget to display when a [child] is triggered.
+  final Widget popup;
+
+  /// An additional constraints to apply to the [popup].
+  final BoxConstraints boxConstraints;
+
+  /// A callback that is called to build the [popup] offset from the [child].
+  final OffsetBuilder offsetBuilder;
+
+  /// A callback that is called to build the route transitions
+  /// to display a [popup].
+  final RouteTransitionsBuilder transitionBuilder;
+
+  /// Duration of the transition going forwards.
+  final Duration transitionDuration;
+
+  /// Indicates whether you can dismiss a [popup] route by tapping the
+  /// modal barrier.
+  final bool barrierDismissible;
+
+  /// A color to use for the [popup] modal barrier.
+  ///
+  /// If this is null, the barrier will be transparent.
+  final Color barrierColor;
+
+  /// A semantic label used for a dismissible barrier when
+  /// a [popup] is triggered.
+  final String barrierLabel;
+
+  /// Creates a new instance of the base popup.
+  ///
+  /// The [barrierDismissible] defaults to `true`.
+  /// If the [boxConstraints] is null, an empty instance of
+  /// the [BoxConstraints] is used.
+  ///
+  /// All the required parameters and the [barrierDismissible] must not be null.
+  const BasePopup({
+    Key key,
+    BoxConstraints boxConstraints,
+    @required this.popup,
+    @required this.offsetBuilder,
+    @required this.transitionBuilder,
+    @required this.child,
+    @required this.transitionDuration,
```

Comment:
Suggest to match order of parameters to the order of the properties of this class - this should improve readability.